### PR TITLE
Add zipkin junit5 module

### PR DESCRIPTION
**Pull request by hezhangjian**

### CODE REVIEW

File: zipkin-junit5/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright 2015-2021 The OpenZipkin Authors
```
Comment (jcchavezs) [inline zipkin-junit5/pom.xml:4]: I guess it should be 2015-2022

Comment (jcchavezs) [inline zipkin-junit5/pom.xml]:
```suggestion
    Copyright 2015-2022 The OpenZipkin Authors
```

File: zipkin-junit5/src/main/java/zipkin2/junit5/HttpFailure.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
```
Comment (jcchavezs) [inline zipkin-junit5/src/main/java/zipkin2/junit5/HttpFailure.java]:
```suggestion
 * Copyright 2015-2022 The OpenZipkin Authors
```

File: zipkin-junit5/src/main/java/zipkin2/junit5/ZipkinDispatcher.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
```
Comment (jcchavezs) [inline zipkin-junit5/src/main/java/zipkin2/junit5/ZipkinDispatcher.java]:
```suggestion
 * Copyright 2015-2022 The OpenZipkin Authors
```

File: zipkin-junit5/src/main/java/zipkin2/junit5/ZipkinExtension.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
```
Comment (jcchavezs) [inline zipkin-junit5/src/main/java/zipkin2/junit5/ZipkinExtension.java]:
```suggestion
 * Copyright 2015-2022 The OpenZipkin Authors
```

File: zipkin-junit5/src/test/java/zipkin2/junit5/ZipkinExtensionTest.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
```
Comment (jcchavezs) [inline zipkin-junit5/src/test/java/zipkin2/junit5/ZipkinExtensionTest.java]:
```suggestion
 * Copyright 2015-2022 The OpenZipkin Authors
```

File: pom.xml [hunk 6]
```diff
@@ -81,13 +82,14 @@
...
     <junit.version>4.13.1</junit.version>
-    <junit-jupiter.version>5.7.0</junit-jupiter.version>
+    <junit-jupiter.version>5.9.0</junit-jupiter.version>
     <mockito.version>3.6.28</mockito.version>
     <assertj.version>3.18.1</assertj.version>
     <awaitility.version>4.0.3</awaitility.version>
     <hamcrest.version>1.3</hamcrest.version>
     <testcontainers.version>1.15.1</testcontainers.version>
     <okhttp.version>4.9.0</okhttp.version>
+    <okhttp5.version>5.0.0-alpha.10</okhttp5.version>
```
Comment (jcchavezs) [inline pom.xml]: Do we want to consume alphas? cc @llinder

Comment (hezhangjian) [reply pom.xml]: currently, okhttp5 only has alpha version, Since we're only using it for testing, I think it's still acceptable

**Comment by Shoothzj** [comment]
@jcchavezs @llinder PTAL again, thanks

**Comment by hezhangjian** [comment]
@jcchavezs @llinder PTAL again, I have changed license years to 2023. thanks

**Comment by jcchavezs** [comment]
@shakuzen do you mind checking this?

**Comment by Shoothzj** [comment]
ping @llinder @shakuzen @jcchavezs

**Comment by hezhangjian** [comment]
ping @llinder @shakuzen @jcchavezs

**Review by shakuzen — COMMENTED** [review]
Overall this looks good, I think. We document things in README files and I don't see one included here. See, for example, the existing [zipkin-junit README](https://github.com/openzipkin/zipkin/blob/master/zipkin-junit/README.md).

**Comment by hezhangjian** [comment]
@shakuzen @jcchavezs Sorry for my late reply. Fixed. PTAL. thanks

**Comment by jcchavezs** [comment]
Thanks, let me run the checks.

**Comment by hezhangjian** [comment]
@jcchavezs I saw the failures in `zipkin-mysql`. what should I do?

**Comment by hezhangjian** [comment]
ping @shakuzen @jcchavezs

**Comment by jcchavezs** [comment]
Checking

File: zipkin-junit5/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
+# zipkin-junit5
+
+This contains `ZipkinRule`, a JUnit rule to spin-up a Zipkin server during tests.
```
Comment (shakuzen) [inline zipkin-junit5/README.md]: This doesn't look right. It contains `ZipkinExtension` which is not a JUnit rule.

**Comment by jcchavezs** [comment]
Please rebase

File: zipkin-junit5/README.md (hunk 7 above)
Comment (hezhangjian) [reply zipkin-junit5/README.md]: @shakuzen Thanks, fixed

**Comment by hezhangjian** [comment]
@jcchavezs Ok. I have done rebasing

**Review by shakuzen — APPROVED** [review]
Looks good to me.

**Comment by hezhangjian** [comment]
ping @jcchavezs Hello, can this pr being merged

**Review by jcchavezs — APPROVED** [review]

**Comment by jcchavezs** [comment]
Thank you and sorry for the delay!

**Comment by jcchavezs** [comment]
We will cut a release very soon

**Pull request merged**